exonum/src/blockchain/tests.rs

```diff
@@ -294,70 +287,63 @@ mod transactions_tests {
     }
 
     impl Transaction for A {
-        fn verify(&self) -> bool {
-            true
-        }
-
-        fn execute(&self, _: &mut Fork) -> ExecutionResult {
+        fn execute(&self, _: TransactionContext) -> ExecutionResult {
             Ok(())
         }
     }
 
     impl Transaction for B {
-        fn verify(&self) -> bool {
-            true
-        }
-
-        fn execute(&self, _: &mut Fork) -> ExecutionResult {
+        fn execute(&self, _: TransactionContext) -> ExecutionResult {
             Ok(())
         }
     }
 
     impl Transaction for C {
-        fn verify(&self) -> bool {
-            true
-        }
-
-        fn execute(&self, _: &mut Fork) -> ExecutionResult {
+        fn execute(&self, _: TransactionContext) -> ExecutionResult {
             Ok(())
         }
     }
 
     #[test]
     fn deserialize_from_json() {
-        fn round_trip<T: Transaction + Serialize>(t: &T) {
+        fn round_trip<T: TransactionSet>(t: &T) {
             let initial = serde_json::to_value(&t).unwrap();
+            println!("{:?}", initial);
             let parsed: MyTransactions = serde_json::from_value(initial.clone()).unwrap();
             let round_tripped = serde_json::to_value(&parsed).unwrap();
             assert_eq!(initial, round_tripped);
         }
 
-        let (_pub_key, sec_key) = gen_keypair();
-        let a = A::new(0, &sec_key);
-        let b = B::new(1, 2, &sec_key);
-        let c = C::new(0, &sec_key);
-        round_trip(&a);
-        round_trip(&b);
-        round_trip(&c);
+        let a = A::new(0);
+        let b = B::new(1, 2);
+        let c = C::new(0);
+        round_trip::<MyTransactions>(&a.into());
+        round_trip::<MyTransactions>(&b.into());
+        round_trip::<MyTransactions>(&c.into());
```

Comment:
Can we avoid turbofishes here? Would it help to add an `Into` trait bound like in `deserialize_from_raw()`?